source/common/http/utility.h

```diff
@@ -676,6 +676,35 @@ getMergedPerFilterConfig(const Http::StreamFilterCallbacks* callbacks,
   return merged;
 }
 
+/**
+ * Return all the available per route filter configs
+ *
+ * @param callbacks The stream filter callbacks to check for route configs.
+ *
+ * @return The all available per route config. The returned pointers are guaranteed to be non-null
+ * and their lifetime is the same as the matched route.
+ */
+template <class ConfigType>
+std::vector<const ConfigType*> getAllPerFilterConfig(const Http::StreamFilterCallbacks* callbacks) {
```

Comment:
Or we make it as a travel function also, then you needn't build a vector anymore

Reply:
We need a vector to stores the returns and RVO (or at worst a move) will make sure there is no additional copy